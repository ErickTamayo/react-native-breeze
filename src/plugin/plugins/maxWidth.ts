import { PluginFunction, PluginPattern } from "./types";

export type PluginGroups = {
  key: string;
};

export const pattern: PluginPattern = ({ keys }) => {
  return new RegExp(`^max-w-(?<key>${keys("maxWidth")})$`);
};

export const plugin: PluginFunction<PluginGroups> = ({ groups, theme }) => {
  const { key } = groups;
  const value = theme(["maxWidth", key]);
  return { maxWidth: value };
};
