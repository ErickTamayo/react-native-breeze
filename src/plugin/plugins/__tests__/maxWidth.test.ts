import {
  generateInput,
  shouldEvaluateTheCorrectPatternTest,
  shouldMatchOutputSnapshot,
} from "../../helpers/tests";
import { pattern, plugin } from "../maxWidth";

describe("maxWidth", () => {
  // prettier-ignore
  const shouldMatch = generateInput('max-w', ['xs','sm',"md",'lg','xl','2xl','3xl','4xl','5xl','6xl']);

  shouldEvaluateTheCorrectPatternTest(pattern, {
    shouldMatch,
    shouldNotMatch: ["max-w-unknown"],
  });

  shouldMatchOutputSnapshot(pattern, plugin, shouldMatch);
});
