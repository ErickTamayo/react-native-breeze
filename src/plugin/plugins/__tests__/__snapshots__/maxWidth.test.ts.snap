// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maxWidth Should parse (max-w-2xl) correctly 1`] = `
Object {
  "maxWidth": 672,
}
`;

exports[`maxWidth Should parse (max-w-3xl) correctly 1`] = `
Object {
  "maxWidth": 768,
}
`;

exports[`maxWidth Should parse (max-w-4xl) correctly 1`] = `
Object {
  "maxWidth": 896,
}
`;

exports[`maxWidth Should parse (max-w-5xl) correctly 1`] = `
Object {
  "maxWidth": 1024,
}
`;

exports[`maxWidth Should parse (max-w-6xl) correctly 1`] = `
Object {
  "maxWidth": 1152,
}
`;

exports[`maxWidth Should parse (max-w-lg) correctly 1`] = `
Object {
  "maxWidth": 512,
}
`;

exports[`maxWidth Should parse (max-w-md) correctly 1`] = `
Object {
  "maxWidth": 448,
}
`;

exports[`maxWidth Should parse (max-w-sm) correctly 1`] = `
Object {
  "maxWidth": 384,
}
`;

exports[`maxWidth Should parse (max-w-xl) correctly 1`] = `
Object {
  "maxWidth": 576,
}
`;

exports[`maxWidth Should parse (max-w-xs) correctly 1`] = `
Object {
  "maxWidth": 320,
}
`;
